independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/ObserverInfo.java

```diff
@@ -123,12 +125,28 @@ int getPriority() {
 
     MethodParameterInfo initEventParam(MethodInfo observerMethod) {
         List<MethodParameterInfo> eventParams = new ArrayList<>();
+
         for (AnnotationInstance annotation : declaringBean.getDeployment().getAnnotations(observerMethod)) {
             if (Kind.METHOD_PARAMETER == annotation.target().kind()
                     && (annotation.name().equals(DotNames.OBSERVES) || annotation.name().equals(DotNames.OBSERVES_ASYNC))) {
                 eventParams.add(annotation.target().asMethodParameter());
+
+                /**
+                 * Log a warning for transactional observers usage since they are not implemented yet:
+                 * https://github.com/quarkusio/quarkus/issues/2224.
+                 */
+                final AnnotationValue value = annotation.value("during");
+                if (value != null) {
+                    if (AnnotationValue.Kind.ENUM.equals(value.kind())
+                            && value.asEnumType().equals(DotNames.TRANSACTION_PHASE)) {
```

Comment:
We should only log a warning if `during != TransactionPhase.IN_PROGRESS` which is the default.

Reply:
Thanks for the review. I have updated this part and the documentation. WDYT?

Reply:
Looks good.